### ISSUE TITLE
breaks with webpack production build

### ISSUE THREAD

**Issue by transcranial** [gka/chroma.js#201]
Having issues with webpack production mode builds (development mode is fine). I've managed to narrow down the issue to code mangling, using a minimal example:

**`index.html`**
```html
<html>
  <head></head>
  <body>
    <script src="main.js"></script>
  </body>
</html>
```

**`index.js`**
```js
const chroma = require('chroma-js')

console.log(
  chroma
    .scale(['#fff', '#000'])(0.5)
    .hex()
)
```

**`webpack.config.js`**
```js
const path = require('path')
const webpack = require('webpack')
const TerserPlugin = require('terser-webpack-plugin')

module.exports = {
  entry: './index.js',
  output: {
    filename: 'main.js',
    path: path.resolve(__dirname)
  },
  optimization: {
    minimizer: [
      new TerserPlugin({
        terserOptions: {
          mangle: true
        }
      })
    ]
  }
}
```

Run: 
```sh
npm install
npm run build
```
and open `index.html` in browser. This is the output in the console:
```
Uncaught Error: unknown format: #4d4d4d
    at new d (main.js:58)
    at b (main.js:58)
    at k (main.js:58)
    at Object.<anonymous> (main.js:1)
    at n (main.js:1)
    at main.js:1
    at main.js:1
```

Setting `mangle: false` in the webpack config, running `npm run build`, the console will simply print `#4d4d4d` without issues. Tweaking the various [mangle options](https://github.com/terser-js/terser#mangle-options) doesn't help either.

Any ideas what is being mangled that's breaking this library?

**Comment by elliotstoner** [comment]
I'm also having this issue... No idea if this is helpful but from what I saw it's trying to auto-detect the format of the args:
```
[
    {
        rgb: [106, 27, 154, 1]
    }
]
```

**Comment by gka** [comment]
hm interesting. 

as background info: over the weekend I ported the entire lib from CoffeeScript to ES6. So now I'm using rollup to compile the build, but what's weird is that even with `rollup-plugin-uglify` I had some issues. While the raw ES6 code would pass all tests, the output of the rollup-uglify (which included mangle) would not.

I "fixed" it by not using the plugin but instead running `uglifyjs` from the standalone npm package after the rollup step. need to do some further digging to see what's causing the mangling to fail..

**Comment by gka** [comment]
ok, this bug seems to be related to the newer versions of uglifyjs, with uglifyjs 2.4 (for instance) it's working fine.

**Comment by gka** [comment]
ah, I think I found the bug. I was using `.constructor.name` to check if a color instance passed to the chroma constructor was already a color. fixed this by just comparing `obj.constructor === this.constructor`.

fixed in 2.0.2
